servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/LoggingHttpLifecycleObserver.java

```diff
@@ -0,0 +1,195 @@
+            requestSize += data.readableBytes();
+        }
+
+        @Override
+        public void onRequestTrailers(final HttpHeaders trailers) {
+            requestTrailers = trailers.size();
+        }
+
+        @Override
+        public void onRequestComplete() {
+            assert requestResult == null;
+            assert requestMetaData != null;
+            requestResult = Result.complete;
+        }
+
+        @Override
+        public void onRequestError(final Throwable cause) {
+            assert requestResult == null;
+            requestResult = cause;
+        }
+
+        @Override
+        public void onRequestCancel() {
+            assert requestResult == null;
+            requestResult = Result.cancelled;
+        }
+
+        @Override
+        public HttpResponseObserver onResponse(final HttpResponseMetaData responseMetaData) {
+            assert this.responseMetaData == null;
+            this.responseMetaData = responseMetaData;
+            return this;
+        }
+
+        @Override
+        public void onResponseData(final Buffer data) {
+            responseSize += data.readableBytes();
+        }
+
+        @Override
+        public void onResponseTrailers(final HttpHeaders trailers) {
+            responseTrailers = trailers.size();
+        }
+
+        @Override
+        public void onResponseComplete() {
+            assert responseResult == null;
+            assert responseMetaData != null;
+            responseResult = Result.complete;
+        }
+
+        @Override
+        public void onResponseError(final Throwable cause) {
+            assert responseResult == null;
+            responseResult = cause;
+        }
+
+        @Override
+        public void onResponseCancel() {
+            assert responseResult == null;
+            responseResult = Result.cancelled;
+        }
+
+        @Override
+        public void onExchangeFinally() {
+            // request info always expected to be available:
+            final HttpRequestMetaData requestMetaData = this.requestMetaData;
+            assert requestMetaData != null;
+            final HttpResponseMetaData responseMetaData = this.responseMetaData;
+            if (responseMetaData != null) {
+                logger.log("connection={} " +
+                        "request=\"{} {} {}\" requestHeaders={} requestSize={} requestTrailers={} requestResult={} " +
+                        "responseCode={} responseHeaders={} responseSize={} responseTrailers={} responseResult={} " +
+                        "duration={}ms",
+                        connInfo == null ? "unknown" : connInfo,
+                        requestMetaData.method(), requestMetaData.requestTarget(), requestMetaData.version(),
+                        requestMetaData.headers().size(), requestSize, requestTrailers, unwrapResult(requestResult),
+                        responseMetaData.status().code(), responseMetaData.headers().size(), responseSize,
```

Comment:
@Scottmitch I remember we discussed keeping only what is necessary instead of the whole `HttpResponseMetaData`, but I reverted it back because I need **deferred** access to headers (more headers can be added later by internal layers). In most cases, the exchange is expected to finish shortly. It should not make any GC difference for most applications.